File: tries/short-encoding-of-words/short-encoding-of-words.js
/**
 * @param {string[]} words
 * @return {number}
 */
function minimumLengthEncoding(words) {
  const wordsIncluded = new Set(words);

  for (const word of words) {
    for (let i = 1; i < word.length; i++) {
      const prefix = word.substring(i);

      if (wordsIncluded.has(prefix)) {
        wordsIncluded.delete(prefix);
      }
    }
  }

  let len = 0;

  for (const word of wordsIncluded) {
    len += word.length + 1;
  }
  return len;
}

File: tries/short-encoding-of-words/short-encoding-of-words.go
package shortencodingofwords

func minimumLengthEncoding(words []string) int {
	wordsIncluded := map[string]bool{}

	for _, word := range words {
		wordsIncluded[word] = true
	}

	for _, word := range words {
		for i := 1; i < len(word); i++ {
			prefix := word[i:]

			if _, ok := wordsIncluded[prefix]; ok {
				delete(wordsIncluded, prefix)
			}
		}
	}

	l := 0

	for word := range wordsIncluded {
		l += len(word) + 1
	}

	return l
}
